tests/layer_tests/pytorch_tests/test_compressed_mm.py

```diff
@@ -0,0 +1,166 @@
+    return decompressed_input
+
+
+def decompress_symmetric(input, scale):
+    input = input.type(dtype=scale.dtype)
+    decompressed_input = input * scale
+    return decompressed_input
+
+
+class TestMatMulU4Weights(PytorchLayerTest):
+    rng = np.random.default_rng(seed=123)
+
+    def _prepare_input(self):
+        return (np.round(5.00 * self.rng.random([2, 4], dtype=np.float32) - 2.50, 4),)
+
+    def create_model(self, group_size):
+        class aten_mm_u4(torch.nn.Module):
+            def __init__(self, compressed_weight, scale, zero_point, weight_shape):
+                super(aten_mm_u4, self).__init__()
+                self.compressed_weight_shape = compressed_weight.shape
+                self.packed_weight = torch.nn.Parameter(pack_uint4(compressed_weight), requires_grad=False)
+
+                self.register_buffer("_scale", scale.type(dtype=torch.float16))
+
+                self.zero_point_shape = zero_point.shape
+                self.register_buffer("_zero_point", pack_uint4(zero_point))
+
+                self.weight_shape = weight_shape
+
+            def forward(self, x):
+                compressed_weight = unpack_uint4(self.packed_weight)
+                compressed_weight = compressed_weight.reshape(self.compressed_weight_shape)
+
+                zero_point = unpack_uint4(self._zero_point)
+                zero_point = zero_point.reshape(self.zero_point_shape)
+
+                weight = decompress_asymmetric(compressed_weight, self._scale, zero_point)
+                weight = weight.reshape(self.weight_shape)
+                weight = weight.type(dtype=torch.float32)
+
+                return torch.matmul(x, weight)
+
+        ref_net = None
+
+        weight_shape = (4, 2)
+        ngroups = weight_shape[0] // group_size
+        compressed_weight_shape = (ngroups, group_size, weight_shape[1])
+        zero_point_shape = scale_shape = (ngroups, 1, weight_shape[1])
+
+        compressed_weight = (15.00 * self.rng.random(compressed_weight_shape, dtype=np.float32)).astype(dtype=np.uint8)
+        scale = np.round(10.00 * self.rng.random(scale_shape, dtype=np.float32) - 5.00)
+        zero_point = (15.00 * self.rng.random(zero_point_shape, dtype=np.float32)).astype(dtype=np.uint8)
+
+        t_compressed_weight = torch.from_numpy(compressed_weight)
+        t_scale = torch.from_numpy(scale)
+        t_zero_point = torch.from_numpy(zero_point)
+
+        return aten_mm_u4(t_compressed_weight, t_scale, t_zero_point, weight_shape), ref_net, ["aten::matmul"]
+
+    @pytest.mark.nightly
+    @pytest.mark.precommit
+    @pytest.mark.precommit_fx_backend
+    @pytest.mark.parametrize("group_size", [2, 4])
+    def test_matmul_u4(self, group_size, ie_device, precision, ir_version):
+        self._test(
+            *self.create_model(group_size),
+            ie_device,
+            precision,
+            ir_version,
+            trace_model=True,
+            dynamic_quantization_group_size=0
```

Comment:
Can we instead of adding a new argument, use existing one, like `quantized_ops`?

Reply:
Thanks for the comment. I thought about this but couldn't find a way to use the existing argument, like `quantized_ops` since it controls the logic of the accuracy check and I need to explicitly disable dynamic quantization and use thresholds for the float operations.